app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -44,6 +44,19 @@ def save_ems_infra_inventory(ems, hashes, target = nil)
       return
     end
 
+    # Check if a vm was removed and we still have disk(s)
+    if target.kind_of?(VmOrTemplate) && hashes[:vms].blank?
+      current = hashes[:vm_disks].collect { |disk| URI(disk[:storage][:ems_ref]).path.split('/').last }.uniq
+      if current.blank?
+        storage = nil
+      else
+        vm_storages = ([target.storage] + target.storages).compact.uniq
+        storage = vm_storages.select { |store| !current.include?(store.ems_ref) }
+      end
+
+      target.disconnect_storage storage
+    end
```

Comment:
Agree this feels like it is in the wrong place.  And now that its changed to target the VM for refresh I don't know that it will even get here.  When I ran this on VMware it disconnected the VM before this new block (here https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh/save_inventory_infra.rb#L43)